### ISSUE TITLE
train_loss= nan train_acc= nan val_loss= nan val 

### ISSUE THREAD

**Issue by monk1337** [yao8839836/text_gcn#2]
Hello,

I am interested in Graph networks and I was reading your paper and came here. I wanted to try your network so after cloning the repo I executed "python remove_words.py" and then "python build_graph.py" but it was taking too much time so I just reduced the no of samples in "20ng.txt" to 4667 samples only for training. After generating all the files "allx,ally,adj etc" I executed python train.py

But I am getting this result:

```
monks-MacBook-Air:text_gcn monk$ python3 train.py
/Users/monk/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
(0, 300) (0,) (4667, 300) (4667, 13) (29744, 300) (29744, 13)
34411
  (0, 84)	7.7466377018461055
  (0, 511)	2.5522340865765614
  (0, 1228)	4.792588675126393
  (0, 1251)	2.3579811682157645
  (0, 1452)	4.1046116522683
  (0, 1578)	3.8522233685584326
  (0, 2288)	5.336421944659694
  (0, 4319)	2.0311063347008216
  (0, 5105)	2.5852514397934945
  (0, 6783)	3.0319339015195625
  (0, 7748)	2.5593122076259895
  (0, 8553)	2.7950449483836333
  (0, 9665)	4.357022284457563
  (0, 10289)	0.7050179417801727
  (0, 10665)	1.2858231481480524
  (0, 11019)	4.368533202923703
  (0, 11427)	1.2107633023653643
  (0, 11476)	2.1480642391734768
  (0, 11503)	5.155688950550878
  (0, 12631)	1.8242683041825598
  (0, 13193)	4.044542925735511
  (0, 13609)	0.12621786067760907
  (0, 13880)	3.2779414152306146
  (0, 13899)	0.9036343000936655
  (0, 14423)	0.6620455766036379
  :	:
  (34410, 20327)	6.838833833515715
  (34410, 20432)	5.1160672357746115
  (34410, 20715)	0.7547901051146402
  (34410, 20766)	2.4443846788432766
  (34410, 21683)	0.6351348001695951
  (34410, 22282)	1.261370725538184
  (34410, 22850)	2.747828172559129
  (34410, 23070)	19.10649061280994
  (34410, 23273)	3.833151229108556
  (34410, 23327)	9.62137117244686
  (34410, 23468)	0.7225003043618639
  (34410, 23634)	3.9596353762176757
  (34410, 23786)	6.838833833515715
  (34410, 24820)	0.23473871655489648
  (34410, 24911)	4.641609256179496
  (34410, 25236)	3.8531518958152255
  (34410, 25877)	0.42964628090406687
  (34410, 26768)	10.540435831203741
  (34410, 26787)	2.4127903134250595
  (34410, 27278)	2.3077888933219883
  (34410, 28249)	2.643136777033327
  (34410, 28311)	3.9924455830251793
  (34410, 28738)	2.7783908229692957
  (34410, 28961)	7.349659457281706
  (34410, 29194)	6.11840003226663
(34411, 34411)
(34411, 34411)
34411
Tensor("graphconvolution_2/SparseTensorDenseMatMul/SparseTensorDenseMatMul:0", shape=(?, 13), dtype=float32)
WARNING:tensorflow:From /Users/monk/Desktop/gra/text_gcn/metrics.py:6: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See tf.nn.softmax_cross_entropy_with_logits_v2.

2018-10-11 12:20:34.296221: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Epoch: 0001 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 65.72113
Epoch: 0002 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 34.03860
Epoch: 0003 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 31.76162
Epoch: 0004 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 33.50022
Epoch: 0005 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 33.36939
Epoch: 0006 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 27.25672
Epoch: 0007 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 26.68194
Epoch: 0008 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 25.18548
Epoch: 0009 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.66044
Epoch: 0010 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.27456
Epoch: 0011 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.75207
Epoch: 0012 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.03114
Epoch: 0013 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 23.75812
Epoch: 0014 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.38517
Epoch: 0015 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.60630
Epoch: 0016 train_loss= nan train_acc= nan val_loss= nan val_acc= nan time= 24.94913
^C^Z
[5]+  Stopped                 python3 train.py
```

Can you let me know is this related to gradient issue loss will be too large ? or something else

**Comment by yao8839836** [comment]
@monk1337 
Hi, thanks for your question.

The reason is that there is no training and validation samples in your 4667 samples. In (0,300), "0" means the real training samples. You may use the first 4667 samples but they are all test examples as indicated in 20ng.txt.

In order to use only a subset of 20NG. you can use the 61-th line in build_graph.py : train_ids = train_ids[:int(0.2 * len(train_ids))]  to select a subset of training data, as the Figure 4 in the paper.

You may also put some of your 4667 test samples to training set by changing their second indicator ''20news-bydate-test" to "20news-bydate-train".

Running 20NG experiment on a MacBook Air will indeed cost much time. I think the easiest way to see some results in the paper is changing  dataset= '20ng' in remove_words.py, build_graph.py and train.py to dataset= 'R8'. Then you can reproduce results for R8 dataset.

**Comment by monk1337** [comment]
Thanks for reply @yao . I am closing the issue.
